docs/src/pages/ComponentArchitecture.mdx

```diff
@@ -0,0 +1,97 @@
+export const meta = {
+  title: 'Component Architecture',
+}
+
+The design and architecture of components should follow from a set of first principles.
+
+- There is 1 right way to do things, and it's easy to do the right thing.
+- Consumers should not be penalized for code they don't use (through size, perf, etc).
+- Don't overgeneralize until we have a clear need (abstracting should solve a bundle size goal, or DRY goal, or partner use case)
+
+# Base components
+
+Each component should be built upon a "base component".
+
+In order to keep sorted lists formatted neatly, each base component should be named with the form: `ComponentNameBase.tsx`
+
+#### Applications which load their own CSS in other means should not pay the penalty of loading a runtime CSS-in-JS package, especially if the class names are known at time of authoring.
+
+Base components should expect a prop called `classes`, which is either a
+dictionary of keys to classnames (where a key represents a portion of the DOM), or
+a callback which will return the same dictionary when called with the component's
+state.
+
+#### Applications which want to override specific aspects of a component should not incur a penalty of loading a concretely referenced component that will not be rendered in the final output.
+
+Base components should expect a prop called `slots`, which is a dictionary of element names mapped
```

Comment:
```suggestion
Components should expect a prop called `slots`, which is a dictionary of element names mapped
```

Another one, should be configured under components or during compose?

Reply:
I think this is true of base components, and compose/createComponents will provide that opinion.

Reply:
Let's leave this as a messy list of "all the requirements" for now.  As we split Fluent UI into more clear layers (base, fluent, `<product>`) we can split these requirements and arch points into "base" and "fluent design" or appropriate.